modules/dataframe/src/jsondataframeconversion.cpp

```diff
@@ -0,0 +1,154 @@
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+#pragma once
+
+#include <inviwo/dataframe/jsondataframeconversion.h>
+#include <inviwo/core/datastructures/buffer/bufferramprecision.h>
+#include <iterator>
+#include <vector>
+
+namespace inviwo {
+
+namespace plot {
+
+namespace detail {
+
+// Helper for adding columns based on json item type
+void addDataFrameColumnHelper(json::value_t valueType, std::string header, DataFrame& df) {
+    switch (valueType) {
+        case json::value_t::null:  ///< null value
+            // need to check more rows to know type
+            break;
+        case json::value_t::object:  ///< object (unordered set of name/value pairs)
+            // Not supported
+            break;
+        case json::value_t::array:  ///< array (ordered collection of values)
+            // Not supported
+            break;
+        case json::value_t::string:  ///< string value
+            df.addCategoricalColumn(header, 0u);
+            break;
+        case json::value_t::boolean:  ///< boolean value
+            // FIXME: Get compile error when trying to add bool column
+            // Error C2039 'data' : is not a member of 'std::vector<bool,std::allocator<_Ty>>' 
+            // \inviwo\core\datastructures\buffer\bufferramprecision.h 155	
+            //df.addColumn<bool>(header, 0u);
+            df.addColumn<int32_t>(header, 0u);
+            break;
+        case json::value_t::number_integer:  ///< number value (signed integer)
+            df.addColumn<int32_t>(header, 0u);
+            break;
+        case json::value_t::number_unsigned:  ///< number value (unsigned integer)
+            df.addColumn<uint32_t>(header, 0u);
+            break;
+        case json::value_t::number_float:  ///< number value (floating-point)
+            df.addColumn<float>(header, 0u);
+        case json::value_t::discarded:  ///< discarded by the the parser callback function
+            break;
+    }
+}
+
+}  // namespace detail
+
+void to_json(json& j, const DataFrame& df) {
+    for (auto row = 0; row < df.getNumberOfRows(); ++row) {
+        json node = json::object();
+        auto items = df.getDataItem(row, true);
+        // Row 0 in the dataframe contains the row indices, which is not needed in the json object.
+        int i = 1;
+        for (auto col = ++items.begin(); col != items.end(); ++col) {
+            node[df.getHeader(i++)] = (*col)->toString();
+        }
+        j.emplace_back(node);
+    }
+}
+
+void from_json(const json& j, DataFrame& df) {
```

Comment:
Might document what  the json should look like...

Reply:
in the header :)